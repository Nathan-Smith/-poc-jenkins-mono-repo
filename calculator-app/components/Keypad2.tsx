import { KeypadButton } from './KeypadButton'

export function Keypad2({
  onClick,
}: {
  onClick?: (op: string) => void
}): JSX.Element {
  return (
    <KeypadButton
      testid="keypad-2"
      gridArea="keypad-2"
      onClick={() => {
        onClick?.('2')
      }}
    >
      2
    </KeypadButton>
  )
}
