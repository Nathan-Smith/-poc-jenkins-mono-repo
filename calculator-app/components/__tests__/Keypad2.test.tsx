import React from 'react'
import renderer, { act, ReactTestRenderer } from 'react-test-renderer'
import { fireEvent, render } from '@testing-library/react'

import { Keypad2 } from '../Keypad2'

describe('Keypad2', () => {
  test('renders', () => {
    let component: ReactTestRenderer | undefined
    act(() => {
      component = renderer.create(<Keypad2 />)
    })

    expect(component?.toJSON()).toMatchSnapshot()
  })

  test('calls onInput with 2 on click', () => {
    const onInput = jest.fn()
    const { getByTestId } = render(<Keypad2 onInput={onInput} />)

    fireEvent.click(getByTestId(/keypad-2/i))

    expect(onInput).toBeCalledWith('2')
  })
})
