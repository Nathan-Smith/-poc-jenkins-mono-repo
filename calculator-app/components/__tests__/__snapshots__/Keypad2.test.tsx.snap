// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Keypad2 renders 1`] = `
<button
  className="mdc-button mdc-button--outlined keypadButton"
  data-testid="keypad-2"
  onClick={[Function]}
  style={
    Object {
      "gridArea": "keypad-2",
    }
  }
>
  <span
    className="mdc-button__ripple"
  />
  <span
    className="mdc-button__label"
  >
    2
  </span>
</button>
`;
